### PR TITLE
Herciuana task2

### PR DESCRIPTION
**Pull request by herciuana**

### CODE REVIEW

File: herciuana_Task2/DynamicStringArray.cpp [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+    }
+    else
+        size--;
+}
+
+string DynamicStringArray::getEntry(const int index) const
+{
+    if ((index < 0) || (index >= size))
+    {
+        return nullptr;
```
Comment (maxlig79) [inline herciuana_Task2/DynamicStringArray.cpp]: nullptr is not a string

File: herciuana_Task2/DynamicStringArray.cpp [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+    for (int i = 0; i < size; i++)
+    {
+        dynamicArray[i] = obj.getEntry(i);
+    }
+}
+
+DynamicStringArray &DynamicStringArray::operator=(const DynamicStringArray &obj)
+{
+    DynamicStringArray temp(obj);
+    if (this->size != 0)
```
Comment (maxlig79) [inline herciuana_Task2/DynamicStringArray.cpp]: This will not work for self assignment.

File: herciuana_Task2/DynamicStringArray.cpp [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+        if (dynamicArray[i] == String)
+        {
+            r = true;
+        }
+    }
+    if (r == true)
+    {
+        return r;
+    }
+    else
```
Comment (maxlig79) [inline herciuana_Task2/DynamicStringArray.cpp]: This mean that if the string is not found the function will not return anything?

File: herciuana_Task2/DynamicStringArray.cpp [hunk 4]
```diff
@@ -0,0 +1,87 @@
+#include "DynamicStringArray.hpp"
+
+DynamicStringArray::DynamicStringArray()
+{
```
Comment (maxlig79) [inline herciuana_Task2/DynamicStringArray.cpp:4]: Please use initialize list, in general you have better performance.

**Review by maxlig79 — CHANGES_REQUESTED** [review]

File: herciuana_Task2/DynamicStringArray.cpp [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+            r = true;
+        }
+    }
+    if (r == true)
+    {
+        return r;
+    }
+    else
+        return size--;
+}
```
Comment (maxlig79) [inline herciuana_Task2/DynamicStringArray.cpp:56]: This is an int and you should return a bool

**Review by maxlig79 — CHANGES_REQUESTED** [review]

File: herciuana_Task2/DynamicStringArray.cpp [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+
+string DynamicStringArray::getEntry(const int index) const
+{
+    if ((index < 0) || (index >= size))
+    {
+        return NULL;
```
Comment (maxlig79) [inline herciuana_Task2/DynamicStringArray.cpp]: The return type is a string, NULL is a macro( zero )

**Review by maxlig79 — CHANGES_REQUESTED** [review]

File: herciuana_Task2/DynamicStringArray.cpp [hunk 7]
```diff
@@ -0,0 +1,110 @@
...
+    bool r = false;
+    int i;
+    for (int i = 0; i < size; i++)
+    {
+        if (dynamicArray[i] == String)
+        {
+            r = true;
+        }
+    }
+    if (r == true)
```
Comment (maxlig79) [inline herciuana_Task2/DynamicStringArray.cpp:37]: Is not easier to return r?
Also this method is incomplete, doesn't delete the entry(string) in the case that is found.

File: herciuana_Task2/test.cpp [hunk 8]
```diff
@@ -0,0 +1,63 @@
...
+BOOST_AUTO_TEST_CASE(test5)
+{
+    DynamicStringArray array6;
+    array6.addEntry("yes");
+    array6.addEntry("no");
+    if (array6.get_Size() != array6.get_Size())
+        BOOST_ERROR("Error");
+}
+BOOST_AUTO_TEST_CASE(test6)
+{
```
Comment (maxlig79) [inline herciuana_Task2/test.cpp:54]: Please add also a test for the move assignment operator

File: herciuana_Task2/test.cpp [hunk 9]
```diff
@@ -0,0 +1,63 @@
...
+    BOOST_CHECK_EQUAL(array2.getEntry(2), "probable");
+}
+BOOST_AUTO_TEST_CASE(test3)
+{
+    DynamicStringArray array3;
+    array3.addEntry("yes");
+    array3.addEntry("no");
+    array3.addEntry("probable");
+    array3.deleteEntry("yes");
+    BOOST_CHECK(array3.get_Size() == 3);
```
Comment (maxlig79) [inline herciuana_Task2/test.cpp]: This test is wrong, the size sould be 2.

**Review by maxlig79 — CHANGES_REQUESTED** [review]

**Review by maxlig79 — DISMISSED** [review]

**Review by silvanadomnica — DISMISSED** [review]

**Review by maxlig79 — APPROVED** [review]

**Review by silvanadomnica — APPROVED** [review]

**Pull request merged**